### ISSUE TITLE
[resnet50] Doesn't work with tensorflow and other things

### ISSUE THREAD

**Issue by vatai** [undertherain/benchmarker#10]
So I've been playing around with benchmarker and tried the two models (vgg16 and resnet50) with the two frameworks I had installed locally (tensorflow and pytorch) and ended up with these results:
|  | resnet50 | vgg16 |
| --- | --- | --- |
| tensorflow | ERROR | looks ok! |
| pytorch | looks ok! | not implemented |

(looks ok! = seems like it's doing what it's supposed to)

**resnet50 with tensorflow**: didn't work! After running:
```shell
python -m benchmarker --framework tensorflow --problem=resnet50 --problem_size=64 --batch_size=8
```
I got an error where the relevant part should be this:
```
Model was constructed with shape (None, 224, 224, 3) for input Tensor("input_1:0", shape=(None, 224, 224, 3), dtype=float32), but it was called on an input with incompatible shape (8, 8, 224, 224, 3).
```
So my guess is that the data generation wrong. IFAIK both resnet and vgg have the same input (224x224) but 

- the [data.py of vgg](https://github.com/vatai/benchmarker/blob/master/benchmarker/modules/problems/vgg16/data.py) uses `get_cubes()` from [util/data/cubes.py](https://github.com/vatai/benchmarker/blob/master/benchmarker/util/data/cubes.py); while
- the [data.py of resnet](https://github.com/vatai/benchmarker/blob/master/benchmarker/modules/problems/resnet50/data.py) uses `gen_images()` from [images_randomized.py](https://github.com/vatai/benchmarker/blob/master/benchmarker/modules/problems/images_randomized.py). 

If I'm not mistaking, one of those two approaches is better(?), which one is it? Any suggestions @undertherain 

**resnet50 on pytorch**: Was OK! (although the sudden jump to 0 loss was kinda strange)
```shell
python -m benchmarker --framework pytorch --problem=resnet50 --problem_size=64 --batch_size=8
```
The strange losses:
```
Train Epoch: 1 [0/8 (0%)]	Loss: 7.656262
Train Epoch: 2 [0/8 (0%)]	Loss: 0.000000
```
**vgg16 on tensorflow**: Was OK!
```shell
python -m benchmarker --framework tensorflow --problem=vgg16 --problem_size=64 --batch_size=8
```
**vgg16 on pytorch**: is not implemented, so I guess this also count as "ok!".

After figuring out how to make resnet work with (at least) tf and pytorch, it shouldn't be too hard to add vgg16 pytorch.

**Comment by undertherain** [comment]
yeah there were some breaking changes
get cubes was an intent to get somewhat patterned inputs as minimal sanity check before moving to say imagenet mini. It was returning all output as one tensor, first dimension being samples. 

the new approach does pre-batched data generation to reduce overhead for inference runs. 

ideally it should be a combination of both :) + option to stream large datasets from storage

**Comment by vatai** [comment]
So `get_cubes()` should (eventually) be depreciated (or merged into `get_images()`), right?

**Comment by undertherain** [comment]
yeah, step 0: use get_images on all imagenet problems
probably need to update do_tensorflow to take data in pre-batched format. the most basic quick and dirty fix - just to stack batched back together into one tensor. Assuming it still used keras's internal interator there

**Comment by vatai** [comment]
Resnet now works both with TF and PyTorch! Yeey! See PR #16. But I still need to 

- update the `data.py` in vgg16 to work with the new type of input tensor,
- just remember maybe the reshape could be done more nicely if i take "keep the last 3" dimensions instead of merging the first 2,
- add pytoch support for vgg,
- and of course tests tests tests!!! (probably a new issue, and maybe a new branch)
